Firstapp/app/src/androidTest/java/com/example/firstapp/ExampleInstrumentedTest.java

```diff
@@ -0,0 +1,160 @@
+package com.example.firstapp;
+
+import android.content.Context;
+import android.content.Intent;
+
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import androidx.test.uiautomator.UiDevice;
+
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObject2;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+import androidx.test.uiautomator.Until;
+
+import java.util.Timer;
+
+import static java.lang.Thread.sleep;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+
+
+import static org.junit.Assert.*;
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class ExampleInstrumentedTest {
+
+    private static final String BASIC_SAMPLE_PACKAGE = "com.example.firstapp";
+    private static final String MAPS_PACKAGE = "com.google.android.apps.maps";
+    private static final int TIMEOUT = 5000;
+    private static final String SEARCH = "Googleplex";
+    private UiDevice device;
+    //target value for searchForTargetNumber()
+    private static Integer TARGET = 2;
+    //Celebration message to be displayed when target is found
+    private static String CELEBRATION = "YAYYYY! " + TARGET + "!!";
+
+
+    private void launchApp(String Package) {
+        // Initialize UiDevice instance
+        device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+        // Start from the home screen
+        device.pressHome();
+
+        // Wait for launcher
+        final String launcherPackage = device.getLauncherPackageName();
```

Comment:
What is the purpose of the launcherPackage?

Reply:
in case pressing the home button lags a bit, we make sure the launcher package appears before launching the app.

Reply:
To simplify things, would it be possible to avoid calling pressHome() and waiting for the launcher?